docs/threat-model.md

```diff
@@ -0,0 +1,54 @@
+# Threat Model
+
+## Overview
+
+netperf is a framework and set of tools for measuring network (and related components) performance. It is designed to be used in a controlled environment, with full control over the systems involved in the test. The primary goal of netperf is to provide accurate and repeatable performance measurements.
+
+### Components
+
+#### netperf Repository
+
+The netperf repository contains the scripts and yaml for the netperf framework and tools. It also contains various secrets to access an Azure Subscription.
+
+#### GitHub Actions
+
+GitHub Actions is used to run the netperf tests. They use the the yaml, scripts and secrets from the netperf repository to run the jobs necessary to perform the tests. These actions are configured to (securely) remotely triggered via [workflow_call](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_call), (securely) manually via [workflow_dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch), or via a netperf Pull Request. Pull Request automation from outside contributors **requires explicit approval**.
+
+#### Azure
+
+Some netperf tests are run on Azure Virtual Machines ([details](machines#azure-virtual-machines)). A single Azure Subscription is used for all resources used in Azure. Generally, these resources are dynamically created and destroyed as needed.
```

Comment:
What secrets are needed for the Azure subscription? I thought federation of the GitHub -> Azure token eliminated the need for explicit secrets.

Reply:
We do use the federated credentials, and we store the `AZURE_CLIENT_ID`, `AZURE_SUBSCRIPTION_ID` and `AZURE_TENANT_ID` in GitHub secrets. I don't know if that's strictly required or not. Do you? Are those not considered secret?